distribution/lib/Standard/AWS/0.0.0-dev/src/Internal/Request_Body.enso

```diff
@@ -0,0 +1,10 @@
+private
+from Standard.Base import all
+from Standard.Base.System.File import file_as_java
+
+polyglot java import software.amazon.awssdk.core.sync.RequestBody
+
+## PRIVATE
+from_local_file (file : File) =
+    java_file = file_as_java file
+    RequestBody.fromFile java_file
```

Comment:
Technically, when we are writing raw bytes from memory (e.g. `Vector.write_bytes`, or encoded `Text.write`), there is no need to create an intermediate file - we could use `RequestBody.fromBytes`/`RequestBody.fromString` which would be a bit more efficient.

However, that would require further customizing the `File_Write_Strategy` logic and I decided to keep things simple, at least for now. So currently all paths go through an `Output_Stream` backed by a temporary file (that is deleted afterwards).

I think we could improve this in the future, to take the simpler path for `Text.write` (as that may be used often I imagine), but I thought it would make sense to do it as part of another ticket, since this one has grown large already.

I'm not sure either how much of a 'performance' difference that would be, given that the biggest cost of the operation is actually contacting the S3 service, so I imagine creating a small temporary file should not be a cost to worry about. 

However, if there's interest in 'improving' this, I'm happy to create a follow-up ticket to avoid the temp file in these few scenarios.

Reply:
Followup: #9022